src/mito/src/table.rs

```diff
@@ -383,22 +435,22 @@ impl<R: Region> MitoTable<R> {
             .await
             .context(UpdateTableManifestSnafu { table_name })?;
 
-        Ok(MitoTable::new(table_info, region, manifest))
+        Ok(MitoTable::new(table_info, regions, manifest))
     }
 
     pub async fn open(
         table_name: &str,
         table_dir: &str,
-        region: R,
+        regions: HashMap<RegionNumber, R>,
         object_store: ObjectStore,
     ) -> Result<MitoTable<R>> {
         let manifest = TableManifest::new(&table_manifest_dir(table_dir), object_store);
 
         let mut table_info = Self::recover_table_info(table_name, &manifest)
             .await?
             .context(TableInfoNotFoundSnafu { table_name })?;
-        table_info.meta.region_numbers = vec![(region.id() & 0xFFFFFFFF) as u32];
-        Ok(MitoTable::new(table_info, region, manifest))
+        table_info.meta.region_numbers = regions.keys().cloned().collect::<Vec<_>>();
```

Comment:
Could we use the region numbers in the table meta instead of requiring the user passing them?

Reply:
I break `MitoTable::open` into `MitoTable::build_manifest`/`MitoTable::recover_table_info`/`MitoTable::new` three parts so we can recover table info (especially regions) without actually open the regions. I also deleted the `stats_regions` related codes, PTAL.